File: mota-service/mota-project-service/src/main/java/com/mota/project/entity/Subtask.java
package com.mota.project.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.mota.common.mybatis.base.BaseEntityDO;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 子任务实体
 * 支持多级子任务结构
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("subtask")
public class Subtask extends BaseEntityDO {

    /**
     * 父任务ID（关联到task表）
     */
    private Long parentTaskId;

    /**
     * 父子任务ID（用于多级子任务，关联到subtask表）
     */
    private Long parentSubtaskId;

    /**
     * 层级深度（0表示一级子任务，1表示二级子任务，以此类推）
     */
    private Integer level;

    /**
     * 所属项目ID
     */
    private Long projectId;

    /**
     * 子任务名称
     */
    private String name;

    /**
     * 子任务描述
     */
    private String description;

    /**
     * 执行人ID
     */
    private Long assigneeId;

    /**
     * 状态(pending/in_progress/completed/cancelled)
     */
    private String status;

    /**
     * 优先级(low/medium/high/urgent)
     */
    private String priority;

    /**
     * 开始日期
     */
    private LocalDate startDate;

    /**
     * 截止日期
     */
    private LocalDate endDate;

    /**
     * 进度(0-100)
     */
    private Integer progress;

    /**
     * 排序顺序
     */
    private Integer sortOrder;

    /**
     * 完成时间
     */
    private LocalDateTime completedAt;

    /**
     * 子子任务列表（非数据库字段）
     */
    @TableField(exist = false)
    private List<Subtask> children;

    /**
     * 子任务数量（非数据库字段）
     */
    @TableField(exist = false)
    private Integer childrenCount;

    /**
     * 已完成子任务数量（非数据库字段）
     */
    @TableField(exist = false)
    private Integer completedChildrenCount;

    /**
     * 状态枚举
     */
    public static class Status {
        public static final String PENDING = "pending";
        public static final String IN_PROGRESS = "in_progress";
        public static final String COMPLETED = "completed";
        public static final String CANCELLED = "cancelled";
    }

    /**
     * 优先级枚举
     */
    public static class Priority {
        public static final String LOW = "low";
        public static final String MEDIUM = "medium";
        public static final String HIGH = "high";
        public static final String URGENT = "urgent";
    }
}

File: mota-web/src/services/api/subtask.ts
import { get, post, put, del } from '../request';

// 子任务类型
export interface Subtask {
  id?: number;
  parentTaskId: number;
  parentSubtaskId?: number;
  level?: number;
  projectId?: number;
  name: string;
  description?: string;
  assigneeId?: number;
  assigneeName?: string;
  assigneeAvatar?: string;
  status?: 'pending' | 'in_progress' | 'completed' | 'cancelled';
  priority?: 'low' | 'medium' | 'high' | 'urgent';
  startDate?: string;
  endDate?: string;
  progress?: number;
  sortOrder?: number;
  completedAt?: string;
  createdAt?: string;
  updatedAt?: string;
  children?: Subtask[];
  childrenCount?: number;
  completedChildrenCount?: number;
}

// 进度汇总类型
export interface SubtaskProgressSummary {
  total: number;
  completed: number;
  inProgress: number;
  pending: number;
  averageProgress: number;
  completionRate: number;
}

// 状态标签
export const SubtaskStatusLabels: Record<string, string> = {
  pending: '待处理',
  in_progress: '进行中',
  completed: '已完成',
  cancelled: '已取消',
};

// 状态颜色
export const SubtaskStatusColors: Record<string, string> = {
  pending: 'default',
  in_progress: 'processing',
  completed: 'success',
  cancelled: 'error',
};

// 优先级标签
export const SubtaskPriorityLabels: Record<string, string> = {
  low: '低',
  medium: '中',
  high: '高',
  urgent: '紧急',
};

// 优先级颜色
export const SubtaskPriorityColors: Record<string, string> = {
  low: 'green',
  medium: 'blue',
  high: 'orange',
  urgent: 'red',
};

// 创建子任务
export const createSubtask = (data: Subtask) => {
  return post<Subtask>('/api/v1/subtasks', data);
};

// 创建子子任务（多级子任务）
export const createChildSubtask = (parentSubtaskId: number, data: Subtask) => {
  return post<Subtask>(`/api/v1/subtasks/${parentSubtaskId}/children`, data);
};

// 批量创建子任务
export const batchCreateSubtasks = (data: Subtask[]) => {
  return post<boolean>('/api/v1/subtasks/batch', data);
};

// 更新子任务
export const updateSubtask = (id: number, data: Partial<Subtask>) => {
  return put<Subtask>(`/api/v1/subtasks/${id}`, data);
};

// 删除子任务（包含所有子子任务）
export const deleteSubtask = (id: number) => {
  return del<boolean>(`/api/v1/subtasks/${id}`);
};

// 获取子任务详情
export const getSubtask = (id: number) => {
  return get<Subtask>(`/api/v1/subtasks/${id}`);
};

// 获取父任务的一级子任务列表
export const getSubtasksByParentTaskId = (parentTaskId: number) => {
  return get<Subtask[]>(`/api/v1/subtasks/parent/${parentTaskId}`);
};

// 获取父任务的所有子任务（包含所有层级）
export const getAllSubtasksByParentTaskId = (parentTaskId: number) => {
  return get<Subtask[]>(`/api/v1/subtasks/parent/${parentTaskId}/all`);
};

// 获取父任务的子任务树形结构
export const getSubtaskTreeByParentTaskId = (parentTaskId: number) => {
  return get<Subtask[]>(`/api/v1/subtasks/parent/${parentTaskId}/tree`);
};

// 获取子任务的子子任务列表
export const getChildSubtasks = (parentSubtaskId: number) => {
  return get<Subtask[]>(`/api/v1/subtasks/${parentSubtaskId}/children`);
};

// 获取项目的所有子任务
export const getSubtasksByProjectId = (projectId: number) => {
  return get<Subtask[]>(`/api/v1/subtasks/project/${projectId}`);
};

// 获取执行人的子任务列表
export const getSubtasksByAssigneeId = (assigneeId: number) => {
  return get<Subtask[]>(`/api/v1/subtasks/assignee/${assigneeId}`);
};

// 更新子任务状态
export const updateSubtaskStatus = (id: number, status: string) => {
  return put<boolean>(`/api/v1/subtasks/${id}/status`, { status });
};

// 更新子任务进度
export const updateSubtaskProgress = (id: number, progress: number) => {
  return put<boolean>(`/api/v1/subtasks/${id}/progress`, { progress });
};

// 完成子任务
export const completeSubtask = (id: number) => {
  return put<boolean>(`/api/v1/subtasks/${id}/complete`);
};

// 分配子任务
export const assignSubtask = (id: number, assigneeId: number) => {
  return put<boolean>(`/api/v1/subtasks/${id}/assign`, { assigneeId });
};

// 移动子任务
export const moveSubtask = (id: number, parentTaskId?: number, parentSubtaskId?: number) => {
  return put<boolean>(`/api/v1/subtasks/${id}/move`, { parentTaskId, parentSubtaskId });
};

// 获取父任务的子任务状态统计
export const getSubtaskStatsByParentTask = (parentTaskId: number) => {
  return get<Record<string, number>>(`/api/v1/subtasks/stats/parent/${parentTaskId}`);
};

// 计算父任务进度
export const calculateParentTaskProgress = (parentTaskId: number) => {
  return get<number>(`/api/v1/subtasks/progress/parent/${parentTaskId}`);
};

// 获取子任务进度汇总
export const getSubtaskProgressSummary = (parentTaskId: number) => {
  return get<SubtaskProgressSummary>(`/api/v1/subtasks/progress/parent/${parentTaskId}/summary`);
};

// 更新子任务排序
export const updateSubtaskSortOrder = (subtasks: Array<{ id: number; sortOrder: number }>) => {
  return put<boolean>('/api/v1/subtasks/sort-order', subtasks);
};